common/src/main/java/com/wynntils/services/itemfilter/SearchQuery.java

```diff
@@ -0,0 +1,162 @@
+                }
+
+                // The filter exists, highlight the keyword...
+                validFilterCharIndices.addAll(
+                        IntStream.rangeClosed(tokenStartIndex, tokenStartIndex + filterString.length())
+                                .boxed()
+                                .toList());
+
+                // ...and try to create the filter only if the filter value is not empty, because a filter without a
+                // value is pointless and we don't want to show the error the factory might return because of that.
+                // We still want to highlight the filter keyword though, that's why we handle the empty value here.
+                if (valueString.isEmpty()) continue;
+                ErrorOr<? extends ItemFilter> filterOrError =
+                        factoryOrError.getValue().create(valueString);
+
+                // If the filter value is invalid, mark the value as ignored and continue to the next token
+                if (filterOrError.hasError()) {
+                    ignoredCharIndices.addAll(IntStream.rangeClosed(
+                                    tokenStartIndex + filterString.length() + 1, tokenStartIndex + token.length())
+                            .boxed()
+                            .toList());
+                    errors.add(filterOrError.getError());
+                    continue;
+                }
+
+                // The filter value is valid, add the filter to the list
+                itemFilters.add(filterOrError.getValue());
+            } else if (!token.isEmpty()) {
+                // The token is not a filter, add it to the list of plain text tokens
+                plainTextTokens.add(token);
+            }
+        }
+
+        return new SearchQuery(
+                queryString, itemFilters, ignoredCharIndices, validFilterCharIndices, errors, plainTextTokens);
+    }
+
+    /**
+     * Checks if the given item matches all filters. Tokens that are not filters in the search query are ignored. If no
+     * filters are present, this method always returns true.
+     *
+     * @param wynnItem the item to check
+     * @return true if the item matches all filters, false otherwise
+     */
+    public boolean filterMatches(WynnItem wynnItem) {
+        return itemFilters.stream().allMatch(o -> o.matches(wynnItem));
+    }
+
+    /**
+     * Checks if the given item name contains the concatenated plain text tokens of the search query. The filter tokens
+     * in the search query are ignored. If there are no plain text tokens, this method always returns true.
+     *
+     * @param itemName the name to check
+     * @return true if the name contains the concatenated plain text tokens, false otherwise
+     */
+    public boolean itemNameMatches(String itemName) {
+        return plainTextTokens.isEmpty()
+                || itemName.toLowerCase(Locale.ROOT)
+                        .contains(String.join(" ", plainTextTokens).toLowerCase(Locale.ROOT));
+    }
```

Comment:
Why not have a generic matches() function and these two as private helpers?

Reply:
In order to match the name of an Item, it has to be an `ItemStack`, whereas the filter matching is done on a `WynnItem`. 
So a `matches()` should take an ItemStack as argument because AFAIK we can't get back an `ItemStack` from a `WynnItem`. But then if the ItemStack does not resolve to a `WynnItem`, what do we do ?

Reply:
> But then if the ItemStack does not resolve to a WynnItem, what do we do?

Return false?